pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -531,6 +533,18 @@ func (c *Controller) Cleanup() error {
 	return nil
 }
 
+func (c *Controller) auditOldServices(services []any) {
+	newServices := make([]*model.Service, 0, len(services))
+	for _, curr := range services {
+		svc := controllers.Extract[*v1.Service](curr)
+		if svc != nil {
+			svcConv := kube.ConvertService(*svc, c.opts.DomainSuffix, c.Cluster())
+			newServices = append(newServices, svcConv)
+		}
+	}
+	c.opts.XDSUpdater.SvcAudit(model.ShardKeyFromRegistry(c), newServices)
```

Comment:
Seems like this compares the services from just one cluster with the services from all clusters? Or am I missing something?

Reply:
The intent was to just audit the svcs from the cluster that was updated.  Thus not wasting any cycles auditing clusters that weren;t updated.  Assuming SyncAll is called cluster by cluster and the shardKey causes per cluster delineation.   I can double check a bit with some testing and confirm.